### PR TITLE
update k8s deps to 18.6 and controller runtime to 0.6.2

### PR DESCRIPTION
**Pull request by mhenriks**
Signed-off-by: Michael Henriksen <mhenriks@redhat.com>

**What this PR does / why we need it**:

Latest!  And greatest?

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

@awels may want to review changes to controller unit tests.  There were some funky things going on there.

**Release note**:

```release-note
k8s deps updated to 18.6 and controller runtime to 0.6.2 
```

### CODE REVIEW

File: hack/build/config.sh [hunk 1]
```diff
@@ -26,8 +26,8 @@ FUNC_TEST_REGISTRY_POPULATE="cdi-func-test-registry-populate"
 FUNC_TEST_REGISTRY_INIT="cdi-func-test-registry-init"
 FUNC_TEST_BAD_WEBSERVER="cdi-func-test-bad-webserver"
 # update this whenever builder Dockerfile is updated
-BUILDER_TAG=${BUILDER_TAG:-0.0.5}
-BUILDER_IMAGE=${BUILDER_IMAGE:-kubevirt/kubevirt-cdi-bazel-builder@sha256:bed4f8c2451747f7f26eedce2b435b7538b9a0350b1dc4e928b005acc4c977a6}
+BUILDER_TAG=${BUILDER_TAG:-0.0.6}
+BUILDER_IMAGE=${BUILDER_IMAGE:-mhenriks/kubevirt-cdi-bazel-builder:0.0.6}
```
Comment (awels) [inline hack/build/config.sh:30]: Need to do the builder dance here, I also have a builder PR up, maybe I can combine what you need there?

File: doc.go [hunk 2]
```diff
@@ -1,21 +0,0 @@
-/*Package root is here for dep dependency manager.
```
[the remark this answers is not in the document]
Comment (mhenriks) [reply doc.go:1]: there had to be at least one go file at root for godep I believe.  Now that one go file is tools.go

File: tools.go [hunk 3]
```diff
@@ -0,0 +1,7 @@
+// +build tools
+
+package tools
+
+import (
```
[the remark this answers is not in the document]
Comment (mhenriks) [reply tools.go:5]: this allows go mod to include the dependencies

File: doc.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (awels) [reply doc.go:1]: Fair enough.

File: tools.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (awels) [reply tools.go:5]: Gotcha

**Comment by awels** [comment]
Looks good to me with the exception of using the wrong builder image, but that should be solved with #1324 when that merges.

**Pull request merged**